projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -118,12 +170,14 @@ public static function get_footer_to_add( $data ) {
 	 * @return string Returns the footer to add for the Stats tracker in a non AMP scenario.
 	 */
 	private static function get_footer( $data ) {
+		$script_attributes = self::get_script_attributes();
+
 		// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
 		// When there is a way to use defer with enqueue, we can move to it and inline the custom data.
```

Comment:
> When there is a way to use defer with enqueue

Isn't that what our `Assets::add_async_script()` does?

I wonder whether switching to that would break the fix for https://github.com/Automattic/jetpack/issues/29560 being made here, or would make it easier since the relevant cookie plugins are hopefully already using `script_loader_tag` to do it generically for all enqueued scripts?

Reply:
> Isn't that what our Assets::add_async_script() does?

It is, yes. We could switch to that soon, once we can safely remove this:
https://github.com/Automattic/jetpack/blob/390e288a9ef8aeac5a074d010fdab506cbd91f3b/projects/plugins/jetpack/modules/stats.php#L157-L167

> I wonder whether switching to that would break the fix for https://github.com/Automattic/jetpack/issues/29560 being made here

I think it would break this PR. We would have to extend `Assets::register_script()` instead, to accept a new option to provide custom script attributes, while taking the existing `async` option into account.

> the relevant cookie plugins are hopefully already using script_loader_tag to do it generically for all enqueued scripts?

I'm afraid we cannot assume that. Since a lot of plugins (Jetpack included in this case) hardcode their scripts so `script_loader_tag` cannot be used, cookie plugins end up having to work around those things (often with JavaScript).

Reply:
> I think it would break this PR.

It would break this PR, yes, but it might also make this PR not be needed at all. If the cookie plugins handle `script_loader_tag` _in addition to_ whatever hacks they need for things like this code.

Reply:
> if the cookie plugins handle script_loader_tag in addition to whatever hacks they need for things like this code.

Looking at https://github.com/fabianpimminger/cookiefox as an example (this is the plugin developed by @fabianpimminger who submitted #29560), I do not see any handling for `script_loader_tag`. @fabianpimminger Could you confirm maybe?

Beyond this, I do not think we can move away from the hardcoded approach where the `get_footer` footer doesn't return a string that can be outputted by `render_footer`. This would break backwards compatibility for `stats_render_footer`. That function is deprecated, but we only deprecated it 5 versions ago. I believe we would want to wait at least one more version to remove it entirely, as per https://github.com/Automattic/jetpack/blob/390e288a9ef8aeac5a074d010fdab506cbd91f3b/projects/plugins/jetpack/functions.global.php#L107-L108

Reply:
Hi @jeherve, thank you very much for your work and the current PR. I am currently working on the integration of various tracking/stats/analytics plugins in my plugin but this code is not public yet. From what I've been seeing, most of these plugins do not use `wp_enqueue_script` and hardcode the scripts similar to the implementation in your current `get_footer` function.

I've looked at this PR and that solution (having a filter for all script attributes) would be ideal from _my_ point of view, but being able to filter the tag via the `script_loader_tag` filter would also be enough to enable these kind of integrations.

Reply:
@fabianpimminger Thanks for the added details!

@anomiex Is something like #29780 what you had in mind?

Reply:
Yes, it is. Thanks!